contracts/service-provider-directory/src/state/nonce.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{constants::SIGNING_NONCES_NAMESPACE, Result};
+
+use cosmwasm_std::{Addr, Storage};
+use cw_storage_plus::Map;
+use nym_contracts_common::signing::Nonce;
+
+pub const NONCES: Map<'_, Addr, Nonce> = Map::new(SIGNING_NONCES_NAMESPACE);
+
+pub fn get_signing_nonce(storage: &dyn Storage, address: Addr) -> Result<Nonce> {
+    let nonce = NONCES.may_load(storage, address)?.unwrap_or(0);
+    Ok(nonce)
+}
+
+pub fn update_signing_nonce(storage: &mut dyn Storage, address: Addr, value: Nonce) -> Result<()> {
```

Comment:
under what conditions would we want to update the nonce to an arbitrary value as opposed to only incrementing it?

Reply:
then I think  it's only used in tests and maybe I forgot to mark it as such

Reply:
Remove the `pub` on the function anyway